### DISCUSSION TITLE
Segmented Control: control activeIndex

### DISCUSSION THREAD

**Discussion by maxdreisbusch** [wix/react-native-ui-lib#2283] [Ideas]
Is it possible to modify the index during runtime? For example I got a formic instance which has the onChangeIndex-Value of the segmented control. When resetting the form, it is not possible to also reset the segmented control to its initial value. Instead it keeps the old selected value.
For that it would be nice to have a value prop where we can control the currently selected index

**Comment by Relax594** [comment]
Is this planned at any time soon? This is such a crucial feature...

@maxdreisbusch did you find any workaround for this?

**Comment by maxdreisbusch** [comment]
Hey @Relax594, we did not find a workaround for that, so we created a custom component consuming rnuilib components by ourselves.

**Comment by MichaelHerrmann1** [comment]
If this gets implemented: It would be great if it's possible to set the controlled index to `-1` (or something else) to enable having no selection at all.

Example use case: We use the SegmentedControl for giving ratings in surveys and want users to make active selections without a default value.
